Project.toml

```diff
@@ -94,7 +94,7 @@ PrecompileTools = "1.1"
 Preferences = "1.3"
 Printf = "1"
 RecipesBase = "1.1"
-RecursiveArrayTools = "2.38.10"
+RecursiveArrayTools = "3.27.1"
```

Comment:
You are setting this compatibility version so high that `SciMLBase` will also adopt a higher version, forcing you to use the new version of the `DiscreteCallback` struct.

That's why even if you're using Julia 1.10 but it still failed here https://github.com/trixi-framework/Trixi.jl/actions/runs/11871869411/job/33084910314?pr=2150#step:7:24

Either change this to a lower version or adopt the new `DiscreteCallback` struct in precompile.